### PR TITLE
Fixed the vstest test failures

### PR DESCRIPTION
**Pull request by cltshivash**

### CODE REVIEW

File: Tasks/VsTest/versionfinder.ts [hunk 1]
```diff
@@ -25,19 +25,19 @@ export class VSTestVersion {
     }
 
     isTestImpactSupported(): boolean {
-        return (this.majorVersion > 15);
+        return (this.majorVersion >= 15);
```
Comment (RanjanarMS) [inline Tasks/VsTest/versionfinder.ts]: Good we got nice bugs in L0 :-)

File: Tests-Legacy/L0/VsTest/_suite.ts [hunk 2]
```diff
@@ -453,10 +453,11 @@ describe('VsTest Suite', function () {
                 assert(tr.succeeded, 'task should have succeeded');
                 assert(tr.ran(vstestCmd), 'should have run vstest');
                 assert(tr.stdout.search(/##vso\[results.publish type=VSTest;mergeResults=false;resultFiles=a.trx;\]/) >= 0, 'should publish test results.');
-                assert(tr.stdout.search(/Install Visual Studio 2015 Update 1 or higher on your build agent machine to run the tests in parallel./) >= 0, 'should have given a warning for update1 or higher requirement');
+                assert(tr.stdout.search(/Install Visual Studio 2015 Update 3 or higher on your build agent machine to run the tests in parallel./) >= 0, 'should have given a warning for update1 or higher requirement');
```
Comment (nigurr) [inline Tests-Legacy/L0/VsTest/_suite.ts]: make it const? using same string in multiple tests

File: Tests-Legacy/L0/VsTest/vs2017.json [hunk 3]
```diff
@@ -27,6 +27,10 @@
         "powershell -NonInteractive -ExecutionPolicy Unrestricted -file /path/to/vs15Helper.ps1": {
             "code": 0,
             "stdout": "<Objs Version=\"1.1.0.1\" xmlns=\"http://schemas.microsoft.com/powershell/2004/04\"><S>\\vs2017\\installation\\folder</S></Objs>"
+        },
+        "wmic datafile where name='\\\\vs2017\\\\installation\\\\folder\\\\Common7\\\\IDE\\\\CommonExtensions\\\\Microsoft\\\\TestWindow\\\\vstest.console.exe' get Version /Value": {
+			"code": 0,
```
Comment (nigurr) [inline Tests-Legacy/L0/VsTest/vs2017.json]: any test for throwing error code?
throwing wrong version string

**Comment by nigurr** [comment]
Squash all commits. These commit messages and file changes are not in sync

File: Tests-Legacy/L0/VsTest/vs2017.json (hunk 3 above)
Comment (cltshivash) [inline Tests-Legacy/L0/VsTest/vs2017.json]: Will take up later. Want to unblock the CI

**Pull request merged**